s3watcher/scripts/properties.js

```diff
@@ -1,12 +1,43 @@
 const fs = require('fs');
 const ini = require('ini');
+const parseHocon = require('hocon-parser');
 
 function load(app) {
     const propertiesFile = `/etc/gu/${app}.properties`;
-    const properties = ini.parse(fs.readFileSync(propertiesFile).toString());
-    return properties;
+    const configFile = `/etc/grid/${app}.conf`;
+    if (fs.existsSync(configFile)) {
+      return {
+        type: "hocon",
+        data: parseHocon(fs.readFileSync(configFile).toString())
+      };
+    }
+    if (fs.existsSync(propertiesFile)) {
+      return {
+        type: "properties",
+        data: ini.parse(fs.readFileSync(propertiesFile).toString())
+      };
+    }
+    console.error(`Neither ${configFile} nor ${propertiesFile} exist`);
+    return {};
+}
+
+function get(config, field) {
+  if (config.type === "properties") {
+    return config.data[field];
+  } else if (config.type === "hocon") {
+    const fieldElements = field.split(".")
+    function getField(obj, fields) {
+      if (fields.length === 0) return obj;
+      getField(obj[fields[0]], fields.slice(1))
+    }
+    return getField(field.data, fieldElements);
+  } else {
+    console.error("Unknown config type")
+    return undefined;
+  }
```

Comment:
minor: as we're returning early, we can omit the `else` to _slightly_ improve readability.

```suggestion
  if (config.type === "properties") {
    return config.data[field];
  }
  if (config.type === "hocon") {
    const fieldElements = field.split(".")
    function getField(obj, fields) {
      if (fields.length === 0) return obj;
      getField(obj[fields[0]], fields.slice(1))
    }
    return getField(field.data, fieldElements);
  }
  console.error("Unknown config type")
  return undefined;
```

Reply:
I'm always in two minds about this. The if ... else structure always seems more explicit in the expression of flow as it's clear that the block will only get executed if the condition doesn't match whilst in the traditional JS style you have to have noticed the return in order to realise that the following code is only executed if the condition doesn't match. I get that this is convention and that my convention comes from Scala but I still prefer it, although not strongly enough if others think I should stick with JS's conventions.

Reply:
Maybe a `switch` would address these concerns? This was a minor observation, feel free to ignore.